java/client/src/test/java/glide/connectors/resources/ThreadPoolResourceAllocatorTest.java

```diff
@@ -0,0 +1,43 @@
+package glide.connectors.resources;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+import io.netty.channel.EventLoopGroup;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Test;
+
+public class ThreadPoolResourceAllocatorTest {
+
+    ThreadPoolResourceAllocator service;
+
+    @Test
+    public void getOrCreateReturnsDefault() {
+        new ThreadPoolResourceAllocator.ShutdownHook().run();
+
+        ThreadPoolResource mockedThreadPool = mock(ThreadPoolResource.class);
+        Supplier<ThreadPoolResource> threadPoolSupplier = mock(Supplier.class);
+        when(threadPoolSupplier.get()).thenReturn(mockedThreadPool);
+
+        ThreadPoolResource theResource = service.getOrCreate(threadPoolSupplier);
+        assertEquals(mockedThreadPool, theResource);
+
+        ThreadPoolResource theSameResource = service.getOrCreate(threadPoolSupplier);
+        assertEquals(mockedThreadPool, theSameResource);
+
+        // and test that the supplier isn't used any longer once the default is set
+        Supplier<ThreadPoolResource> notUsedSupplier = mock(Supplier.class);
+        ThreadPoolResource firstResource = service.getOrCreate(notUsedSupplier);
+        verify(notUsedSupplier, times(0)).get();
```

Comment:
why isn't this assert on line 30, on threadPoolSupplier?

Reply:
I'm not certain which assertion you're referencing, but the purpose of the verification here is to ensure that, following the initial setup, the supplier function is not executed again. This is because we've already established the shared resource. I have made the test more concise where it still achieves the same functionality.